README.rst

```diff
@@ -1,186 +1,160 @@
-the package.
+   $ pip install jupyterlab
+   $ jupyter-lab
 
-**IMPORTANT:** if the user provides a custom installation of GridTools
-C++ sources, it should be compatible with the latest stable release of
-GridTools, which is the version targeted by GT4Py. Note that a
-compilation problem may also appear if there is a different GridTools
-C++ version installed in a standard prefix (e.g. ``/usr/local``) which
-could be included in user-provided or standard ``setuptools`` include
-paths (for example if *Boost* is installed in the same prefix).
+There are two options to run the unit and integration tests in ``tests/``:
 
-As usual in Python, we strongly recommended to create a new virtual
-environment for any project:
+1. Use tox: ``pip install tox && tox -r -e py38-all-cpu``.
+2. Install the development requirements: ``pip install -r requirements-dev.txt``,
+   then ``pytest`` can execute the tests directly.
 
-::
 
-    # Create a virtual environment using the 'venv' module
-    python -m venv path_for_the_new_venv
+📖 Description
+--------------
 
-    # Activate the virtual environment and make sure that 'wheel' is installed
-    source path_for_the_new_venv/bin/activate
-    pip install --upgrade wheel
+GT4Py is a Python library for expressing finite-difference stencils
+related to weather and climate modeling using a high-level
+DSL using Python functions.
+These functions are compiled by the framework into high-performance
+implementations for CPUs and GPUs.
 
-Recommended installation for regular users
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The DSL expresses the stencils using the parallel model from the
+`GridTools C++ Framework <https://github.com/GridTools/gridtools>`__,
+and uses it, as well as other backends, for optimized code generation.
+In this Cartesian parallel model there are always three dimensions,
+and the vertical (``K``) is treated separately from the horizontal axes
+(``I``, ``J``), which are iterated over in parallel.
 
-If you are not planning to modify GT4Py sources, the easiest way to
-complete the installation would be:
+The *gtscript* DSL expresses Cartesian offsets from a center index, as it
+would be written algorithmically.
+List of these stencil *statements* form *computations*, and can be specialized
+in the vertical index to account for boundaries, or accumulate fields.
 
-::
+In order to execute stencils, arrays are required.
+For this, GT4Py has functions which create storages for the given stencil backend.
+The following backends are supported:
 
-    # First, clone the repository
-    git clone https://github.com/gridtools/gt4py.git
+- ``numpy``: Pure-Python backend
+- ``gt:cpu_ifirst``: GridTools C++ CPU backend using ``I``-first data ordering
+- ``gt:cpu_kfirst``: GridTools C++ CPU backend using ``K``-first data ordering
+- ``gt:gpu``: GridTools backend for CUDA
+- ``cuda``: CUDA backend minimally using utilities from GridTools
```

Comment:
Some love for dace? :D

Reply:
Yes, we should make sure those are added before merging. I consider this PR still quite far from mergable, so there's time :)